ext/kv/sqlite.rs

```diff
@@ -111,18 +112,27 @@ impl<P: SqliteDbHandlerPermissions> DatabaseHandler for SqliteDbHandler<P> {
     path: Option<String>,
   ) -> Result<Self::DB, AnyError> {
     let conn = match (path.as_deref(), &self.default_storage_dir) {
-      (Some(":memory:") | None, None) => {
+      (Some(":memory:"), _) | (None, None) => {
         rusqlite::Connection::open_in_memory()?
       }
       (Some(path), _) => {
+        if path.is_empty() {
+          return Err(type_error("Filename cannot be empty"));
+        }
+        if path.starts_with(':') {
+          return Err(type_error(
+            "Filename cannot start with ':' unless prefixed with './'",
+          ));
+        }
```

Comment:
https://www.sqlite.org/c3ref/open.html
> Future versions of SQLite might make use of additional special filenames that begin with the ":" character. It is recommended that when a database filename actually does begin with a ":" character you should prefix the filename with a pathname such as "./" to avoid ambiguity.